sensor: adxl345: Decoder-related bug-fixes

ubieda opened the pull request:
### Description
This PR applies various fixes to the ADXL345 driver:
- Decoder:
	- Add `adxl345_get_size_info()` used, for instance, by the Sensor Shell.
	- Fix  `adxlXXX_decode_sample()` to use `sensor_three_axis_data`.
	- Fix Q-scale factor scaling tables, and add some description on the calculation rationale.
- Driver:
	- Disable Streaming by default so users do not need to worry about it unless explicitly interested in such functionality.
	- Only write FIFO_STREAMING config-bit if using FIFO, to suppress delay in sample acquisition for basic read/decode applications.

Fixes #84252 

### Testing
Run Shell sample with sensor commands on the ADXL345 (tested both for SPI and I2C). The following evidence is for I2C:
``` console
west build -b nrf52840dk/nrf52840 samples/subsys/shell/shell_module
```
- DTS Overlay (`samples/subsys/shell/shell_module/boards/nrf52840dk_nrf52840.overlay`)
``` device-tree
/ {
	aliases {
		accel0 = &adxl345;
	};
};

&arduino_i2c {
	status = "okay";
	
	adxl345: adxl345@53 {
		compatible = "adi,adxl345";
		reg = <0x53>;
		status = "okay";
		int2-gpios = <&gpio1 3 GPIO_ACTIVE_HIGH>;
	};
};
```
- Config Overlay (`samples/subsys/shell/shell_module/boards/nrf52840dk_nrf52840.conf`)
```
CONFIG_SENSOR=y
CONFIG_SENSOR_ASYNC_API=y
CONFIG_SENSOR_SHELL=y
```
- Sample Output:
``` console
uart:~$ sensor get adxl345@53 
channel type=3(accel_xyz) index=0 shift=7 num_samples=1 value=8177551269ns, (0.000000, -0.459686, 8.887274)
uart:~$ sensor get adxl345@53
channel type=3(accel_xyz) index=0 shift=7 num_samples=1 value=8825378417ns, (0.000000, -0.459686, 8.887274)
uart:~$ sensor get adxl345@53
channel type=3(accel_xyz) index=0 shift=7 num_samples=1 value=9273559570ns, (0.000000, -0.459686, 8.887274)
uart:~$ sensor get adxl345@53
channel type=3(accel_xyz) index=0 shift=7 num_samples=1 value=9713378906ns, (0.000000, -0.459686, 8.887274)
uart:~$ sensor get adxl345@53
channel type=3(accel_xyz) index=0 shift=7 num_samples=1 value=10129577636ns, (0.000000, -0.459686, 9.040503)
uart:~$ sensor get adxl345@53
channel type=3(accel_xyz) index=0 shift=7 num_samples=1 value=12177551269ns, (-1.532288, -0.612915, 9.346961)
uart:~$ sensor get adxl345@53
channel type=3(accel_xyz) index=0 shift=7 num_samples=1 value=12641601562ns, (1.685517, -6.129155, 4.137179)
uart:~$ 

```

ubieda committed:
sensor: adxl345: Add get_size_info API

Used by the sensor-shell in order to retrieve values, otherwise
it crashes.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda committed:
sensor: adxl345: Fix decoder for non-streaming mode

The following fixes have been applied to this decoder:
- The Q-scale factor was fixed, both for full-scale and non
full-scale modes.
- The data-type decoded is struct sensor_three_axis_data, as
it should for read/decode API.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

cfriedt commented on drivers/sensor/adi/adxl345/adxl345_decoder.c:
```diff
@@ -208,17 +208,20 @@ static int adxl345_decode_sample(const struct adxl345_sample *data,
-		adxl345_accel_convert(out++, data->x);
-		adxl345_accel_convert(out++, data->y);
-		adxl345_accel_convert(out, data->z);
+		adxl345_accel_convert_q31(out->readings->x, data->x, data->selected_range, false);
+		adxl345_accel_convert_q31(out->readings->y, data->y, data->selected_range, false);
+		adxl345_accel_convert_q31(out->readings->z, data->z, data->selected_range, false);
 		break;
```
```suggestion
		adxl345_accel_convert_q31(&out->readings->x, data->x, data->selected_range, false);
		adxl345_accel_convert_q31(&out->readings->y, data->y, data->selected_range, false);
		adxl345_accel_convert_q31(&out->readings->z, data->z, data->selected_range, false);
		break;
```

ubieda committed:
sensor: adxl345: Only enable FIFO Stream with Sensor Stream is enabled

Otherwise with its default configuration (25-Hz, 32-level FIFO),
getting individual samples could be up to 1-second old.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda committed:
sensor: adxl345: Disable Sensor Streaming by default

Have the application enable this feature explicitcly, so that
simple applications do not need to disable this to get the
expected behavior.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda commented:
Just rebased. Need to create a GH issue to get this backported into 4.0.0

laurensslats commented:
Tested and it works for me using the ADXL345.

ubieda commented:
Created issue and added backport label

cfriedt commented:
I also tested this on the `xiao_esp32c3` with `adxl343`

MaureenHelm reviewed (approved):
Thanks @ubieda !

I tested this out with the following and it looks good:
```
$ west build -p -b apard32690//m4 --shield pmod_acl samples/subsys/shell/shell_module/ -- -DCONFIG_SENSOR=y -DCONFIG_SENSOR_ASYNC_API=y -DCONFIG_SENSOR_SHELL=y
```

@dimitrije-lilic please take a look
